### ISSUE TITLE
NIRSpec Stage 6 error with lsq

### ISSUE THREAD

**Issue by jovian-explorer** [kevin218/Eureka#325]
### Instrument

NIRSpec (Stages 1-3), Other (any stage)

### What happened?

NIRSpec Stage 6 is showing some error. The thing is that I haven't changed anything in my stage 5 ecf and epf, except the fitting method as lsq. But, somehow the stage 5 ran. Now the issue is with stage 6.

### Error traceback output

    Starting Stage 6: Light Curve Fitting
    
    Input directory: /home/keshav/Data/JWST-Sim/NIRSpec/Stage5/S5_2022-05-17_wasp39b_run1/ap6_bg7/
    Output directory: /home/keshav/Data/JWST-Sim/NIRSpec/Stage6/S6_2022-05-17_wasp39b_run1/ap6_bg7/
    Copying S6 control file
    Traceback (most recent call last):
      File "/home/keshav/Data/JWST-Sim/NIRSpec/run_eureka.py", line 31, in <module>
        s6_meta = s6.plot_spectra(eventlabel, ecf_path=ecf_path, s5_meta=s5_meta)
      File "/home/keshav/miniconda3/lib/python3.9/site-packages/eureka/S6_planet_spectra/s6_spectra.py", line 154, in plot_spectra
        lower = np.abs((meta.spectrum_median-meta.spectrum_err[0,:])**2-meta.spectrum_median**2)
    IndexError: too many indices for array: array is 0-dimensional, but 2 were indexed

### What operating system are you using?

linux ubuntu 20.04

### What version of Python are you running?

Python 3.9.7

### What Python packages do you have installed?

    # packages in environment at /home/keshav/miniconda3:
    #
    # Name                    Version                   Build  Channel
    _libgcc_mutex             0.1                        main  
    _openmp_mutex             4.5                       1_gnu  
    alabaster                 0.7.12                   pypi_0    pypi
    asdf                      2.10.1                   pypi_0    pypi
    asdf-astropy              0.2.1                    pypi_0    pypi
    asdf-coordinates-schemas  0.1.0                    pypi_0    pypi
    asdf-standard             1.0.2                    pypi_0    pypi
    asdf-transform-schemas    0.2.2                    pypi_0    pypi
    asdf-wcs-schemas          0.1.1                    pypi_0    pypi
    asteval                   0.9.26                   pypi_0    pypi
    astraeus                  0.1                      pypi_0    pypi
    astropy                   5.0.4                    pypi_0    pypi
    astropy-healpix           0.6                      pypi_0    pypi
    astroquery                0.4.6                    pypi_0    pypi
    astroscrappy              1.1.0                    pypi_0    pypi
    asttokens                 2.0.5                    pypi_0    pypi
    attrs                     21.4.0                   pypi_0    pypi
    babel                     2.10.1                   pypi_0    pypi
    backcall                  0.2.0                    pypi_0    pypi
    batman-package            2.4.8                    pypi_0    pypi
    bayesicfitting            3.0.1                    pypi_0    pypi
    beautifulsoup4            4.11.1                   pypi_0    pypi
    bleach                    5.0.0                    pypi_0    pypi
    bokeh                     2.4.2                    pypi_0    pypi
    bottleneck                1.3.4                    pypi_0    pypi
    brotlipy                  0.7.0           py39h27cfd23_1003  
    ca-certificates           2021.10.26           h06a4308_2  
    ccdproc                   2.3.1                    pypi_0    pypi
    celerite                  0.4.2                    pypi_0    pypi
    certifi                   2021.10.8        py39h06a4308_2  
    cffi                      1.15.0           py39hd667e15_1  
    cftime                    1.6.0                    pypi_0    pypi
    charset-normalizer        2.0.4              pyhd3eb1b0_0  
    cloudpickle               2.0.0                    pypi_0    pypi
    commonmark                0.9.1                    pypi_0    pypi
    conda                     4.11.0           py39h06a4308_0  
    conda-content-trust       0.1.1              pyhd3eb1b0_0  
    conda-package-handling    1.7.3            py39h27cfd23_1  
    corner                    2.2.1                    pypi_0    pypi
    crds                      11.14.0                  pypi_0    pypi
    cryptography              36.0.0           py39h9ce1e76_0  
    cycler                    0.11.0                   pypi_0    pypi
    cython                    0.29.28                  pypi_0    pypi
    dask                      2022.5.0                 pypi_0    pypi
    decorator                 5.1.1                    pypi_0    pypi
    defusedxml                0.7.1                    pypi_0    pypi
    docutils                  0.17.1                   pypi_0    pypi
    drizzle                   1.13.4                   pypi_0    pypi
    dynesty                   1.2.2                    pypi_0    pypi
    emcee                     3.1.2                    pypi_0    pypi
    entrypoints               0.4                      pypi_0    pypi
    eureka                    0.0.1                    pypi_0    pypi
    executing                 0.8.3                    pypi_0    pypi
    fastjsonschema            2.15.3                   pypi_0    pypi
    filelock                  3.6.0                    pypi_0    pypi
    fonttools                 4.33.3                   pypi_0    pypi
    fsspec                    2022.3.0                 pypi_0    pypi
    future                    0.18.2                   pypi_0    pypi
    george                    0.4.0                    pypi_0    pypi
    gwcs                      0.18.1                   pypi_0    pypi
    h5netcdf                  1.0.0                    pypi_0    pypi
    h5py                      3.1.0                    pypi_0    pypi
    html5lib                  1.1                      pypi_0    pypi
    idna                      3.3                pyhd3eb1b0_0  
    imageio                   2.19.1                   pypi_0    pypi
    imagesize                 1.3.0                    pypi_0    pypi
    importlib-metadata        4.11.3                   pypi_0    pypi
    iniconfig                 1.1.1                    pypi_0    pypi
    ipython                   8.3.0                    pypi_0    pypi
    jedi                      0.18.1                   pypi_0    pypi
    jeepney                   0.8.0                    pypi_0    pypi
    jinja2                    3.1.2                    pypi_0    pypi
    jmespath                  1.0.0                    pypi_0    pypi
    jsonschema                3.2.0                    pypi_0    pypi
    jupyter-client            7.3.1                    pypi_0    pypi
    jupyter-core              4.10.0                   pypi_0    pypi
    jupyterlab-pygments       0.2.2                    pypi_0    pypi
    jwst                      1.3.3                    pypi_0    pypi
    keyring                   23.5.0                   pypi_0    pypi
    kiwisolver                1.4.2                    pypi_0    pypi
    ld_impl_linux-64          2.35.1               h7274673_9  
    libffi                    3.3                  he6710b0_2  
    libgcc-ng                 9.3.0               h5101ec6_17  
    libgomp                   9.3.0               h5101ec6_17  
    libstdcxx-ng              9.3.0               hd4cf53a_17  
    lmfit                     1.0.3                    pypi_0    pypi
    locket                    1.0.0                    pypi_0    pypi
    lxml                      4.8.0                    pypi_0    pypi
    markdown-it-py            2.1.0                    pypi_0    pypi
    markupsafe                2.1.1                    pypi_0    pypi
    matplotlib                3.5.2                    pypi_0    pypi
    matplotlib-inline         0.1.3                    pypi_0    pypi
    mdit-py-plugins           0.3.0                    pypi_0    pypi
    mdurl                     0.1.1                    pypi_0    pypi
    mistune                   0.8.4                    pypi_0    pypi
    myst-parser               0.17.2                   pypi_0    pypi
    nbclient                  0.6.3                    pypi_0    pypi
    nbconvert                 6.5.0                    pypi_0    pypi
    nbformat                  5.4.0                    pypi_0    pypi
    nbsphinx                  0.8.8                    pypi_0    pypi
    ncurses                   6.3                  h7f8727e_2  
    nest-asyncio              1.5.5                    pypi_0    pypi
    netcdf4                   1.5.8                    pypi_0    pypi
    networkx                  2.8                      pypi_0    pypi
    numpy                     1.22.3                   pypi_0    pypi
    numpydoc                  1.3.1                    pypi_0    pypi
    openssl                   1.1.1m               h7f8727e_0  
    packaging                 21.3                     pypi_0    pypi
    pandas                    1.4.2                    pypi_0    pypi
    pandocfilters             1.5.0                    pypi_0    pypi
    parsley                   1.3                      pypi_0    pypi
    parso                     0.8.3                    pypi_0    pypi
    partd                     1.2.0                    pypi_0    pypi
    pexpect                   4.8.0                    pypi_0    pypi
    photutils                 1.1.0                    pypi_0    pypi
    pickleshare               0.7.5                    pypi_0    pypi
    pillow                    9.1.0                    pypi_0    pypi
    pip                       21.2.4           py39h06a4308_0  
    pluggy                    1.0.0                    pypi_0    pypi
    poppy                     1.0.2                    pypi_0    pypi
    prompt-toolkit            3.0.29                   pypi_0    pypi
    psutil                    5.9.0                    pypi_0    pypi
    ptyprocess                0.7.0                    pypi_0    pypi
    pure-eval                 0.2.2                    pypi_0    pypi
    py                        1.11.0                   pypi_0    pypi
    pycosat                   0.6.3            py39h27cfd23_0  
    pycparser                 2.21               pyhd3eb1b0_0  
    pyerfa                    2.0.0.1                  pypi_0    pypi
    pygments                  2.12.0                   pypi_0    pypi
    pyopenssl                 21.0.0             pyhd3eb1b0_1  
    pyparsing                 3.0.9                    pypi_0    pypi
    pyrsistent                0.18.1                   pypi_0    pypi
    pysocks                   1.7.1            py39h06a4308_0  
    pysynphot                 2.0.0                    pypi_0    pypi
    pytest                    7.1.2                    pypi_0    pypi
    python                    3.9.7                h12debd9_1  
    python-dateutil           2.8.2                    pypi_0    pypi
    pytz                      2022.1                   pypi_0    pypi
    pyvo                      1.3                      pypi_0    pypi
    pywavelets                1.3.0                    pypi_0    pypi
    pyyaml                    6.0                      pypi_0    pypi
    pyzmq                     22.3.0                   pypi_0    pypi
    readline                  8.1.2                h7f8727e_1  
    recommonmark              0.7.1                    pypi_0    pypi
    reproject                 0.8                      pypi_0    pypi
    requests                  2.27.1             pyhd3eb1b0_0  
    ruamel_yaml               0.15.100         py39h27cfd23_0  
    scikit-image              0.19.2                   pypi_0    pypi
    scipy                     1.8.0                    pypi_0    pypi
    secretstorage             3.3.2                    pypi_0    pypi
    semantic-version          2.9.0                    pypi_0    pypi
    setuptools                58.0.4           py39h06a4308_0  
    six                       1.16.0             pyhd3eb1b0_0  
    snowballstemmer           2.2.0                    pypi_0    pypi
    soupsieve                 2.3.2.post1              pypi_0    pypi
    spherical-geometry        1.2.22                   pypi_0    pypi
    sphinx                    4.5.0                    pypi_0    pypi
    sphinx-rtd-theme          1.0.0                    pypi_0    pypi
    sphinxcontrib-applehelp   1.0.2                    pypi_0    pypi
    sphinxcontrib-devhelp     1.0.2                    pypi_0    pypi
    sphinxcontrib-htmlhelp    2.0.0                    pypi_0    pypi
    sphinxcontrib-jsmath      1.0.1                    pypi_0    pypi
    sphinxcontrib-qthelp      1.0.3                    pypi_0    pypi
    sphinxcontrib-serializinghtml 1.1.5                    pypi_0    pypi
    sqlite                    3.37.0               hc218d9a_0  
    stack-data                0.2.0                    pypi_0    pypi
    stcal                     0.6.4                    pypi_0    pypi
    stdatamodels              0.2.4                    pypi_0    pypi
    stpipe                    0.3.3                    pypi_0    pypi
    stsci-image               2.3.5                    pypi_0    pypi
    stsci-imagestats          1.6.3                    pypi_0    pypi
    stsci-stimage             0.2.5                    pypi_0    pypi
    svo-filters               0.4.1                    pypi_0    pypi
    tifffile                  2022.5.4                 pypi_0    pypi
    tinycss2                  1.1.1                    pypi_0    pypi
    tk                        8.6.11               h1ccaba5_0  
    tomli                     2.0.1                    pypi_0    pypi
    toolz                     0.11.2                   pypi_0    pypi
    tornado                   6.1                      pypi_0    pypi
    tqdm                      4.62.3             pyhd3eb1b0_1  
    traitlets                 5.2.0                    pypi_0    pypi
    tweakwcs                  0.7.4                    pypi_0    pypi
    typing-extensions         4.2.0                    pypi_0    pypi
    tzdata                    2021e                hda174b7_0  
    uncertainties             3.1.6                    pypi_0    pypi
    urllib3                   1.26.7             pyhd3eb1b0_0  
    wcwidth                   0.2.5                    pypi_0    pypi
    webencodings              0.5.1                    pypi_0    pypi
    wheel                     0.37.1             pyhd3eb1b0_0  
    xarray                    2022.3.0                 pypi_0    pypi
    xz                        5.2.5                h7b6447c_0  
    yaml                      0.2.5                h7b6447c_0  
    zipp                      3.8.0                    pypi_0    pypi
    zlib                      1.2.11               h7f8727e_4  

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jovian-explorer** [comment]
Stage 6 ecf - 
    
    # Eureka! Control File for Stage 6: Spectra Plotting
    
    # Stage 6 Documentation: https://eurekadocs.readthedocs.io/en/latest/ecf.html#stage-6
    
    allapers        False # Run S6 on all of the apertures considered in S5? Otherwise will use newest output in the inputdir
    
    # Plotting parameters
    y_unit          (Rp/Rs)^2 # Options include Rp/Rs, (Rp/Rs)^2, Fp/Fs
    y_scalar        100 # Can be used to convert to percent (100), ppm (1e6), etc.
    x_unit          um # Options include any measurement of light included in astropy.units.spectral (e.g. um, nm, Hz, etc.)
    
    # This section is relevant if isplots_S6>=3
    # Scale height parameters (if you want a second copy of the plot with a second y-axis with units of scale height)
    star_Rad        0.6506 # The radius of the star in units of solar radii
    planet_Teq      1400 # The equilibrium temperature of the planet in units of Kelvin
    planet_Mass     2 # The planet's mass in units of Jupiter masses (used to calculate surface gravity)
    planet_Rad      None # The planet's radius in units of Jupiter radii (used to calculate surface gravity); Set to None to use the average fitted radius
    planet_mu       2.3 # The mean molecular mass of the atmosphere (in atomic mass units)
    planet_R0       None # The reference radius (in Jupiter radii) for the scale height measurement; Set to None to use the mean fitted radius
    
    # Diagnostics
    isplots_S6      5 # Generate few (1), some (3), or many (5) figures (Options: 1 - 5)
    hide_plots      False # If True, plots will automatically be closed rather than popping up
    
    # Project directory
    topdir          /home/keshav/Data/JWST-Sim/NIRSpec
    
    # Model to plot underneath the fitted data points
    model_spectrum  None # Path to the model spectrum relative to topdir. Set to None if no model should be plotted.
    model_x_unit    um # Options include any measurement of light included in astropy.units.spectral (e.g. um, nm, Hz, etc.)
    model_y_unit    Rp/Rs # Options include Rp/Rs, (Rp/Rs)^2, Fp/Fs
    model_y_scalar  1 # Indicate whether model y-values have already been scaled (e.g. write 1e6 if model_spectrum is in ppm)
    model_delimiter None # Delimiter between columns. Typical options: None (for space), ',' for comma
    
    # Directories relative to topdir
    inputdir        /Stage5  # The folder containing the outputs from Eureka!'s S5 pipeline (will be overwritten if calling S5 and S6 sequentially)
    outputdir       /Stage6

**Comment by jovian-explorer** [comment]
Stage 5 ecf - 
         
    # Eureka! Control File for Stage 5: Lightcurve Fitting
    
    # Stage 5 Documentation: https://eurekadocs.readthedocs.io/en/latest/ecf.html#stage-5
    
    ncpu            8 # The number of CPU threads to use when running emcee or dynesty in parallel
    
    allapers        False                   # Run S5 on all of the apertures considered in S4? Otherwise will use newest output in the inputdir
    rescale_err     False                   # Rescale uncertainties to have reduced chi-squared of unity
    fit_par         ./S5_fit_par_template.epf   # What fitting epf do you want to use?
    verbose         True                    # If True, more details will be printed about steps
    fit_method      [lsq]               #options are: lsq, emcee, dynesty (can list multiple types separated by commas)
    run_myfuncs     [batman_tr,polynomial]  #options are: batman_tr, batman_ecl, sinusoid_pc, expramp, GP, and polynomial (can list multiple types separated by commas)
    
    # Limb darkening controls (not yet implemented)
    #fix_ld          False #use limb darkening file?
    #ld_file         /path/to/limbdarkening/ld_outputfile.txt  #location of limb darkening file
    
    # General fitter
    old_fitparams   None # filename relative to topdir that points to a fitparams csv to resume where you left off (set to None to start from scratch)
    
    #lsq
    lsq_method      'Nelder-Mead' # The scipy.optimize.minimize optimization method to use
    lsq_tol         1e-6    # The tolerance for the scipy.optimize.minimize optimization method
    lsq_maxiter     None    # Maximum number of iterations to perform. Depending on the method each iteration may use several function evaluations. Set to None to use the default value
    
    #mcmc
    old_chain       None    # Output folder relative to topdir that contains an old emcee chain to resume where you left off (set to None to start from scratch)
    lsq_first       True    # Initialize with an initial lsq call (can help shorten burn-in, but turn off if lsq fails). Only used if old_chain is None
    run_nsteps      1000
    run_nwalkers    200
    run_nburn       500     # How many of run_nsteps should be discarded as burn-in steps
    
    #dynesty
    run_nlive       1024    # Must be > ndim * (ndim + 1) // 2
    run_bound       'multi'
    run_sample      'auto'
    run_tol         0.1
    
    #GP inputs
    kernel_inputs   ['time'] #options: time
    kernel_class    ['Matern32'] #options: ExpSquared, Matern32, Exp, RationalQuadratic for george, Matern32 for celerite (sums of kernels possible for george separated by commas)
    GP_package      'celerite' #options: george, celerite
    
    # Plotting controls
    interp          False   # Should astrophysical model be interpolated (useful for uneven sampling like that from HST)
    
    # Diagnostics
    isplots_S5      5       # Generate few (1), some (3), or many (5) figures (Options: 1 - 5)
    testing_S5      False   # Boolean, set True to only use the first spectral channel
    testing_model   False   # Boolean, set True to only inject a model source of systematics
    hide_plots      False   # If True, plots will automatically be closed rather than popping up
    
    # Project directory
    topdir          /home/keshav/Data/JWST-Sim/NIRSpec
    
    # Directories relative to topdir
    inputdir        /Stage4   # The folder containing the outputs from Eureka!'s S4 pipeline (will be overwritten if calling S4 and S5 sequentially)
    outputdir       /Stage5


Stage 5 epf - 

      # Stage 5 Fit Parameters Documentation: https://eurekadocs.readthedocs.io/en/latest/ecf.html#stage-5-fit-parameters
      
      
      #Name        Value         Free?          PriorPar1    PriorPar2    PriorType
      # PriorType can be U (Uniform), LU (Log Uniform), or N (Normal).
      # If U/LU, PriorPar1 and PriorPar2 represent upper and lower limits of the parameter/log(the parameter).
      # If N, PriorPar1 is the mean and PriorPar2 is the standard deviation of a Gaussian prior.
      #-------------------------------------------------------------------------------------------------------
      #
      # ------------------
      # ** Transit/eclipse parameters **
      # ------------------
      rp           0.173010       'free'         0.1         0.3          U
      #fp           0.008         'free'         0            0.5          U
      # ----------------------
      # ** Phase curve parameters **
      # ----------------------
      #AmpCos1      0.4           'free'         0            1            U
      #AmpSin1      0.01          'free'         -1           1            U
      #AmpCos2      0.01          'free'         -1           1            U
      #AmpSin2      0.01          'free'         -1           1            U
      # ------------------
      # ** Orbital parameters **
      # ------------------
      per          4.055259   	'free'         4.055259   	1e-5		 N
      t0           0.5		 	'free'         0.4     0.6     U
      time_offset  0            	'independent'
      inc          87.93        	'free'         85       90        U
      a            11.612         'free'         10         13         U
      ecc          0.0           	'fixed'        0            1            U
      w            90.           	'fixed'        0            180          U
      # -------------------------
      # ** Limb darkening parameters **
      # Choose limb_dark from ['uniform', 'linear', 'quadratic', 'kipping2013', 'square-root', 'logarithmic', 'exponential', '4-parameter']
      # -------------------------
      limb_dark    'kipping2013'	'independent'
      u1           0.5           	'free'         0            1            U
      u2           0.5           	'free'         0            1            U
      # --------------------
      # ** Systematic variables **
      # polynomial model variables (c0--c9 for 0th--3rd order polynomials in time); Fitting at least c0 is very strongly recommended!
      # expramp model variables (r0--r2 for one exponential ramp, r3--r5 for a second exponential ramp)
      # GP model parameters (A, WN, m1, m2) in log scale
      # --------------------
      c0           1             	'free'         0.95         1.05         U
      c1           0           	'free'         -0.1         0.1          U
      #A            -5           	'free'         -20.         -5.          U
      #WN           -15          	'free'         -20.         -10.         U
      #m1           -5          	'free'         -10.         0.           U
      # -----------
      # ** White noise **
      # Use scatter_mult to fit a multiplier to the expected noise level from Stage 3 (recommended)
      # Use scatter_ppm to fit the noise level in ppm
      # Use WN parameter in GP model parameters to account for white noise in the GP model
      # -----------
      scatter_mult 1             	'free'         1            0.1          N

**Comment by kevin218** [comment]
The problem is likely with meta.spectrum_err. Running lsq doesn't generate uncertainties, so the parameter is probably empty. Consider running emcee in S5, plotting the spectrum yourself, or modifying the code to catch the error.

**Comment by jovian-explorer** [comment]
I ran emcee in S5, and S6 ran without a hitch as well. But, as I switched back to dynesty, S6 again failed.

    Traceback (most recent call last):
      File "/home/keshav/Data/JWST-Sim/NIRSpec/run_eureka.py", line 29, in <module>
        s5_meta = s5.fitlc(eventlabel, ecf_path=ecf_path, s4_meta=s4_meta)
      File "/home/keshav/miniconda3/lib/python3.9/site-packages/eureka/S5_lightcurve_fitting/s5_fit.py", line 173, in fitlc
        meta = fit_channel(meta,time,flux,channel,flux_err,eventlabel,sharedp,params,log,longparamlist,time_units,paramtitles,chanrng)
      File "/home/keshav/miniconda3/lib/python3.9/site-packages/eureka/S5_lightcurve_fitting/s5_fit.py", line 257, in fit_channel
        lc_model.fit(model, meta, log, fitter='dynesty')
      File "/home/keshav/miniconda3/lib/python3.9/site-packages/eureka/S5_lightcurve_fitting/lightcurve.py", line 156, in fit
        fit_model = self.fitter_func(self, model, meta, log, **kwargs)
      File "/home/keshav/miniconda3/lib/python3.9/site-packages/eureka/S5_lightcurve_fitting/fitters.py", line 644, in dynestyfitter
        plots.plot_corner(samples, lc, meta, freenames, fitter='dynesty')
      File "/home/keshav/miniconda3/lib/python3.9/site-packages/eureka/S5_lightcurve_fitting/plots_s5.py", line 203, in plot_corner
        fig = corner.corner(samples, fig=fig, quantiles=[0.16, 0.5, 0.84], max_n_ticks=3,
      File "/home/keshav/miniconda3/lib/python3.9/site-packages/corner/corner.py", line 226, in corner
        return corner_impl(
      File "/home/keshav/miniconda3/lib/python3.9/site-packages/corner/core.py", line 121, in corner_impl
        raise ValueError(
    ValueError: It looks like the parameter(s) in column(s) 9 have no dynamic range. Please provide a `range` argument.

**Comment by kevin218** [comment]
Sounds like you should stick with emcee for now.

**Comment by taylorbell57** [comment]
Ah okay, I remember having once encountered that same issue as you in Stage 6 with lsq - I thought I'd written code to take care of that, but apparently not enough. I can look into it more and will report back here.

You should open a new issue about that problem with dynesty though, but before you do that I'd first recommend looking at all the S5 plots from your lsq fit, your emcee fit, and what plots were made (if any) from your dynesty fit (and any logged messages during the dynesty fit) to see if you can troubleshoot the issue as it may well not be an issue with eureka but with your ECF or EPF settings.

**Comment by jovian-explorer** [comment]
Okay sure, I'll open a new issue for the problem with dynesty. But, the thing is I haven't changed anything in the ecf and the epf files since the last time.

**Comment by taylorbell57** [comment]
Reopening this issue to remind me to patch the bug you found in Stages 6 with lsq results
